### PR TITLE
Trim unnecessary trailing zeroes in data segments

### PR DESCRIPTION
**Pull request by dcodeIO**
After https://github.com/AssemblyScript/assemblyscript/pull/1586 it may not be that useful anymore to keep the exact contents of compiler-generated managed data segments in untouched output (for manual inspection), due to explicitly aligning up to the next valid block size and filling with even more zeroes than before. Perhaps trimming is more useful now?

- [x] I've read the contributing guidelines

### CODE REVIEW

File: tests/compiler/memory.untouched.wat [hunk 1]
```diff
@@ -4,25 +4,25 @@
...
+ (data (i32.const 110) "")
+ (data (i32.const 111) "")
+ (data (i32.const 112) "")
+ (data (i32.const 128) "")
+ (data (i32.const 144) "")
+ (data (i32.const 152) "")
+ (data (i32.const 156) "")
+ (data (i32.const 158) "")
+ (data (i32.const 159) "")
+ (data (i32.const 160) "")
```
Comment (MaxGraey) [inline tests/compiler/memory.untouched.wat:22]: Could such empty data segments also avoided after trimming?

Comment (dcodeIO) [reply tests/compiler/memory.untouched.wat:22]: Hmm, good question. In this case we'd also lose the visual info that there is a data segment starting there. Still unsure how useful this or the zeroes are when manually viewing compiler output, somewhat depending on how common it is to run untouched binaries, and how far we'd like to go in terms of exactly displaying what becomes compiled.

Comment (MaxGraey) [reply tests/compiler/memory.untouched.wat:22]: > Hmm, good question. In this case we'd also lose the visual info that there is a data segment starting there

Good point

**Comment by dcodeIO** [comment]
Closing in favor of clarity.

**Pull request closed without merging**